include/sound/compress_driver.h

```diff
@@ -180,19 +187,29 @@ static inline void snd_compr_drain_notify(struct snd_compr_stream *stream)
 
 /**
  * snd_compr_set_runtime_buffer - Set the Compress runtime buffer
- * @substream: compress substream to set
+ * @stream: compress stream to set
  * @bufp: the buffer information, NULL to clear
  *
  * Copy the buffer information to runtime buffer when @bufp is non-NULL.
  * Otherwise it clears the current buffer information.
  */
-static inline void snd_compr_set_runtime_buffer(
-					struct snd_compr_stream *substream,
-					struct snd_dma_buffer *bufp)
+static inline void
+snd_compr_set_runtime_buffer(struct snd_compr_stream *stream,
+			     struct snd_dma_buffer *bufp)
 {
-	struct snd_compr_runtime *runtime = substream->runtime;
-
-	runtime->dma_buffer_p = bufp;
+	struct snd_compr_runtime *runtime = stream->runtime;
```

Comment:
just wondering if we need to check if (runtime) as well?

Reply:
Based on snd_pcm_set_runtime_buffer.
However, if you look at the code of core kernel api, you will notice the lack of checks, ifs and whatnot in many cases. This is because kernel internal api requires programmer to be sane about his/her doings enabling further optimization in the process. _set_runtime_buffer is certainly not exposed to usespace nor is fed with any data coming from it. Because of that, the choice for leaving said function without runtime check was (and still is) the right choice.